framework/config/src/main/java/org/apache/cloudstack/framework/config/impl/ConfigDepotImpl.java

```diff
@@ -268,6 +277,48 @@ public ConfigurationDao global() {
         return _configDao;
     }
 
+    protected String getConfigStringValueInternal(String cacheKey) {
+        String[] parts = cacheKey.split("-");
+        String key = parts[0];
+        ConfigKey.Scope scope = ConfigKey.Scope.Global;
+        Long scopeId = null;
+        try {
+            scope = ConfigKey.Scope.valueOf(parts[1]);
+            scopeId = Long.valueOf(parts[2]);
+        } catch (IllegalArgumentException ignored) {}
+        if (!ConfigKey.Scope.Global.equals(scope) && scopeId != null) {
+            ScopedConfigStorage scopedConfigStorage = null;
+            for (ScopedConfigStorage storage : _scopedStorages) {
+                if (storage.getScope() == scope) {
+                    scopedConfigStorage = storage;
+                }
+            }
+            if (scopedConfigStorage == null) {
+                throw new CloudRuntimeException("Unable to find config storage for this scope: " + scope + " for " + key);
```

Comment:
Should we fail here? Or get the value from the DB in this case?

Reply:
This is inline with the exiting `ConfigDepotImpl.findScopedConfigStorage`. Also, it may then allow returning config value for undesired scopes